fastcrypto-zkp/src/circom.rs

```diff
@@ -3,20 +3,100 @@
 
 use ark_bn254::{Fq, Fq2, G1Affine, G1Projective, G2Affine, G2Projective};
 use fastcrypto::error::FastCryptoError;
-
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
 /// A G1 point in BN254 serialized as a vector of three strings which is the canonical decimal
 /// representation of the projective coordinates in Fq.
-pub type CircomG1 = Vec<String>;
+pub type CircomG1 = Vec<Bn254Point>;
 
 /// A G2 point in BN254 serialized as a vector of three vectors each being a vector of two strings
 /// which are the canonical decimal representation of the coefficients of the projective coordinates
 /// in Fq2.
-pub type CircomG2 = Vec<Vec<String>>;
+pub type CircomG2 = Vec<Vec<Bn254Point>>;
+
+#[derive(Debug, Clone, JsonSchema, Eq, PartialEq)]
+pub struct Bn254Point([u8; 32]);
```

Comment:
This should be called Bn254FieldElement or Bn254FqElement or something like that.

Reply:
Also, I think it could be a wrapper around a `Fq` element instead of a byte array.